source/simulator/stokes_matrix_free.cc

```diff
@@ -857,13 +795,18 @@ namespace aspect
 
     for (unsigned int cell=cell_range.first; cell<cell_range.second; ++cell)
       {
-        const VectorizedArray<number> &cell_one_over_viscosity = one_over_viscosity(cell);
+        // The /= operator for VectorizedArray results in a foating point operation
+        // (divide by 0) since the (*viscosity)(cell) array is not completely filled.
+        // Therefore, we need to divide each entry manually.
+        VectorizedArray<number> &one_over_cell_viscosity = (*viscosity)(cell);
```

Comment:
you are taking a reference to the vectorized array, doesn't this modify the content in there?

Reply:
You're right, it should. Not sure why this wasn't messing up the tests.. Maybe I was lucky with the order of operations.. I've changed this.